### DISCUSSION TITLE
Argument of type '"/_layout/child"' is not assignable to parameter of type 'keyof FileRoutesByPath'.

### DISCUSSION THREAD

**Discussion by wong0220** [TanStack/router#2962] [Q&A]
Looking at the official document, I'm making a child file wrapped in layout, and the code keeps getting the same error as the picture, can you tell me what the problem is?
<img width="266" alt="스크린샷 2024-12-09 오후 11 24 25" src="https://github.com/user-attachments/assets/df4f0f24-ec73-4fdc-9237-aa852dd3410d">
<img width="859" alt="스크린샷 2024-12-09 오후 11 24 31" src="https://github.com/user-attachments/assets/f6b693a0-b738-45f1-84b4-76655d1964a1">
<img width="889" alt="스크린샷 2024-12-09 오후 11 24 39" src="https://github.com/user-attachments/assets/b0ac9952-4e02-4e53-82ee-13d3d400ef56">

**Comment by schiller-manuel** [comment]
sometimes the routeTree.gen.ts file is open and needs to be closed/reopened for TS to properly parse it.

**Reply by lavizp** [reply]
Such an simple solution, Thanks.

**Reply by Kadajett** [reply]
> sometimes the routeTree.gen.ts file is open and needs to be closed/reopened for TS to properly parse it.

Yeah, thank you lol That did it.

**Reply by muhammad-paksi** [reply]
Closing the routetree doesn't solve it. Once I checked the routetree.gen.ts, and something surprising showed up. There's a syntax error in the file which was auto generated. This problem has been around for two years since I saw a user brought up a similar problem (the not assignable problem) on Stack Overflow in 2024.
